File: xrecycleviewlib/src/main/java/com/ashlikun/xrecycleview/nested/NestedInnerScrollView.java
package com.ashlikun.xrecycleview.nested;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.widget.NestedScrollView;

/**
 * 作者　　: 李坤
 * 创建时间: 2019/12/13　11:17
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：RecyclerView里面嵌套ViewPager的时候的内部ScrollView
 */
public class NestedInnerScrollView extends NestedScrollView {
    protected NestedOnChildTouch childTouch = new NestedOnChildTouch(this);

    public NestedInnerScrollView(@NonNull Context context) {
        super(context);
    }

    public NestedInnerScrollView(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public NestedInnerScrollView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public boolean onTouchEvent(MotionEvent e) {
        if (childTouch.onTouchEvent(e)) {
            return false;
        }
        return super.onTouchEvent(e);
    }
}

File: xrecycleviewlib/src/main/java/com/ashlikun/xrecycleview/nested/NestedInnerScrollView.kt
package com.ashlikun.xrecycleview.nested

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.core.widget.NestedScrollView

/**
 * @author　　: 李坤
 * 创建时间: 2021/12/9 22:26
 * 邮箱　　：496546144@qq.com
 *
 * 功能介绍：RecyclerView里面嵌套ViewPager的时候的内部ScrollView
 */

open class NestedInnerScrollView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyle: Int = 0
) : NestedScrollView(context, attrs, defStyle) {
    protected var childTouch = NestedOnChildTouch(this)

    override fun onTouchEvent(e: MotionEvent): Boolean {
        return if (childTouch.onTouchEvent(e)) {
            false
        } else super.onTouchEvent(e)
    }
}